modules/nlu/src/backend/engine.ts

```diff
@@ -439,6 +451,51 @@ export default class ScopedEngine implements Engine {
       .digest('hex')
   }
 
+  public async validateIntentSlots(intent: sdk.NLU.IntentDefinition, lang: string): Promise<IntentValidation> {
```

Comment:
I think that this has nothing to do in engine as it has absolutely nothing related to training or prediction. We might want to create some sort of validation service outside engine. Further more, engine is under serious rework at the moment so this will cause major conflicts.